db/python/tables/output_file.py

```diff
@@ -0,0 +1,266 @@
+from google.cloud.storage import Client
+
+from db.python.tables.base import DbBase
+from models.models.output_file import OutputFileInternal, RecursiveDict
+
+
+class OutputFileTable(DbBase):
+    """
+    Capture Analysis table operations and queries
+    """
+
+    table_name = 'output_file'
+
+    async def process_output_for_analysis(
+        self, analysis_id: int, output: str | None, outputs: str | RecursiveDict | None
+    ):
+        """
+        Process output for analysis
+        """
+        if output and outputs:
+            warnings.warn(
+                'The output field is deprecated, using outputs instead since it was passed in..',
+                PendingDeprecationWarning,
+                2,
+            )
+        if output and not outputs:
+            warnings.warn(
+                'The output field is deprecated, please use outputs instead',
+                PendingDeprecationWarning,
+                2,
+            )
+
+        if outputs and isinstance(outputs, str):
+            warnings.warn(
+                'The outputs field should be a dictionary, passing a str will be deprecated soon.',
+                PendingDeprecationWarning,
+                2,
+            )
+
+        output_data = outputs or output
+
+        if output_data:
+            await self.create_or_update_analysis_output_files_from_output(
+                analysis_id=analysis_id, json_dict=output_data
+            )
+
+    async def create_or_update_output_file(
+        self,
+        path: str,
+        parent_id: int | None = None,
+        client: Client | None = None,
+    ) -> int | None:
+        """
+        Create a new file, and add it to database
+        """
+        file_obj = AnyPath(path, client=GSClient(storage_client=client))
+
+        if not file_obj or not client:
+            raise ValueError('Invalid file path or client')
+
+        file_info = await OutputFileInternal.get_file_info(
+            file_obj=file_obj, client=client
+        )
+
+        if not file_info or not file_info.get('valid'):
+            return None
+
+        kv_pairs = [
+            ('path', path),
+            ('basename', file_info['basename']),
+            ('dirname', file_info['dirname']),
+            ('nameroot', file_info['nameroot']),
+            ('nameext', file_info['nameext']),
+            ('file_checksum', file_info['checksum']),
+            ('size', file_info['size']),
+            ('valid', file_info['valid']),
+            ('parent_id', parent_id),
+        ]
+
+        kv_pairs = [(k, v) for k, v in kv_pairs if v is not None]
+        keys = [k for k, _ in kv_pairs]
+        cs_keys = ', '.join(keys)
+        cs_id_keys = ', '.join(f':{k}' for k in keys)
+        non_pk_keys = [k for k in keys if k != 'path']
+        update_clause = ', '.join(
+            [f'{k} = VALUES({k})' for k in non_pk_keys]
+        )  # ON DUPLICATE KEY UPDATE {update_clause}
+
+        async with self.connection.transaction():
```

Comment:
No need for a transaction if there's only one query

Reply:
Oh and this is already happening in a transaction which is started in the `create_or_update_analysis_output_files_from_output` method, meaning this would be creating checkpoints which I don't think are needed